src/ReturnTypes/AuthManagerExtension.php

```diff
@@ -49,11 +68,66 @@ public function getTypeFromMethodCall(
             return ParametersAcceptorSelector::selectSingle($methodReflection->getVariants())->getReturnType();
         }
 
-        return TypeCombinator::addNull(
-            TypeCombinator::union(...array_map(
-                static fn (string $authModel): Type => new ObjectType($authModel),
-                $authModels,
-            )),
-        );
+        $type = TypeCombinator::union(...array_map(
+            static fn (string $authModel): Type => new ObjectType($authModel),
+            $authModels,
+        ));
+
+        return $type;
+    }
+
+    public function getTypeFromUserCall(
+        MethodReflection $methodReflection,
+    ): Type {
+        return TypeCombinator::addNull($this->getTypeFromAuthenticateCall($methodReflection));
+    }
+
+    public function getTypeFromGuardCall(
+        MethodCall $methodCall,
+        Scope $scope,
+    ): Type {
+        $defaultReturnType = TypeCombinator::intersect(new ObjectType(Guard::class), new ObjectType(StatefulGuard::class));
+
+        $config = $this->getContainer()->get('config');
+        if ($config === null) {
+            return $defaultReturnType;
+        }
+
+        /** @var array<string, mixed> $guards */
+        $guards = $config->get('auth.guards');
+        /** @var string $defaultGuard */
+        $defaultGuard = $config->get('auth.defaults.guard');
+
+        if (count($methodCall->getArgs()) === 0) {
+            if (! array_key_exists($defaultGuard, $guards)) {
+                return $defaultReturnType;
+            }
+
+            return $this->findTypeFromGuardDriver($guards[$defaultGuard]['driver']) ?? $defaultReturnType;
+        }
+
+        $argType    = $scope->getType($methodCall->getArgs()[0]->value);
+        $argStrings = $argType->getConstantStrings();
+
+        if (count($argStrings) !== 1) {
+            return $defaultReturnType;
+        }
+
+        $driver = $config->get('auth.guards.' . $argStrings[0]->getValue() . '.driver', null);
+        if (! is_string($driver)) {
+            return $defaultReturnType;
+        }
+
+        return $this->findTypeFromGuardDriver($driver) ?? $defaultReturnType;
+    }
+
+    private function findTypeFromGuardDriver(string $driver): Type|null
+    {
+        return match ($driver) {
+            'session' => new ObjectType('Illuminate\Auth\SessionGuard'),
```

Comment:
This can also be a `::class` string?

Reply:
Yes, I hadn't spotted that when moving it from src/ReturnTypes/GuardDynamicStaticMethodReturnTypeExtension.php